KEP-3344: Add Release Team Roster KEP (#3347)
kubernetes/enhancements: keps/sig-release/3344-refine-release-team/README.md

-->

[kubernetes.io]: https://kubernetes.io/
[kubernetes/enhancements]: https://git.k8s.io/enhancements
[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
[kubernetes/website]: https://git.k8s.io/website

## Summary

<!--
This section is incredibly important for producing high-quality, user-focused
documentation such as release notes or a development roadmap. It should be
possible to collect this information before implementation begins, in order to
avoid requiring implementors to split their attention between writing release
notes and implementing the feature itself. KEP editors and SIG Docs
should help to ensure that the tone and content of the `Summary` section is
useful for a wide audience.

A good summary is probably at least a paragraph in length.

Both in this section and below, follow the guidelines of the [documentation
style guide]. In particular, wrap lines to a reasonable length, to make it
easier for reviewers to cite specific portions, and to minimize diff churn on
updates.

[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
-->

This change will introduce a "Release Team Roster".

> This roster will be a list of contributors, maintained by SIG Releases, who wish to contribute to current or future release teams.

Reviewer: ```suggestion
This roster will be a list of contributors, maintained by SIG Release, who wish to contribute to current or future release teams.
```

- The release team is not "stable", as continuing requires constant re-application through the shadow survey.
  This inhibits the ability of contributors to gain comfort and skill in a particular role. 
- Wider community members interact with a constantly changing team of people for every release, which makes it harder for them to build working relationships with the release team.
- With a few notable exceptions, once a contributor ends their time on the release team they don't come back.
  This "up or out" process makes it hard to do release team work as a long-standing community role.

There are a number of qualities to the current process that are desirable, and should be kept:

- The shadow survey is an excellent "on ramp" into the community, providing contributors with no experience a path to [org membership](https://github.com/kubernetes/community/blob/master/community-membership.md) and it is expected that contributors may only join the release team for a release or two before moving to other parts of the community.
- The release team having some degree of 'churn' is natural and expected.
  An entirely static, or effectively static, team is not desirable. 

> Balancing these desires, this KEP introduces a stable

Reviewer: I think this sentence needs an ending? ;)

Author: I just replaced the whole sentance. I don't know what thought I was half way though. 😂

- Change how the release team lead, emeritus adviser, and role leads are selected.

## Proposal

<!--
This is where we get down to the specifics of what the proposal actually is.
This should have enough detail that reviewers can understand exactly what
you're proposing, but should not include things like API designs or
implementation. What is the desired outcome and how do we measure success?.
The "Design Details" section below is for the real
nitty-gritty.
-->

### Release Team Roster

We will add to SIG Release a "Release Team Roster".
The list of roster members will be maintained in a YAML file in https://github.com/kubernetes/sig-release under a new release-team/roster directory.
People on the roster can be in one of three roles: Shadow, Member, or Emeritus.

At the time of creation, all former role leads or release leads from any release will be added as emeritus members, or a member if they want.
The 1.25 EA will lead discussions around which shadows from 1.25 should be placed at the member or shadow level on the release team roster.

Roster Shadows may leave at any time by opening a PR to remove themselves.

> Shadows that do not join a release team for two consecutive releases will generally be removed (although this is at the discression of SIG Release).

Reviewer: ```suggestion
Shadows that do not join a release team for two consecutive releases will generally be removed (although this is at the discretion of SIG Release).
```

in the form of:
- <package>: <date> - <current test coverage>
The data can be easily read from:
https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit

This can inform certain test coverage improvements that we want to do before
extending the production code to implement this enhancement.
-->

- `<package>`: `<date>` - `<test coverage>`
Not applicable.

##### Integration tests

<!--
This question should be filled when targeting a release.
For Alpha, describe what tests will be added to ensure proper quality of the enhancement.

For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
https://storage.googleapis.com/k8s-triage/index.html
-->
Not applicable.

##### e2e tests

<!--
This question should be filled when targeting a release.
For Alpha, describe what tests will be added to ensure proper quality of the enhancement.

For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
https://storage.googleapis.com/k8s-triage/index.html

We expect no non-infra related flakes in the last month as a GA graduation criteria.
-->

Not applicable.

### Graduation Criteria

> 

Reviewer: This is essentially a straight to GA process change right?

Author: Good point. I agree. I've updated it.

I don't think there's any way to do an alpha/beta of this.

  - [Failure mode brief description]
    - Detection: How can it be detected via metrics? Stated another way:
      how can an operator troubleshoot without logging into a master or worker node?
    - Mitigations: What can be done to stop the bleeding, especially for already
      running user workloads?
    - Diagnostics: What are the useful log messages and their required logging
      levels that could help debug the issue?
      Not required until feature graduated to beta.
    - Testing: Are there any tests for failure mode? If not, describe why.
-->

###### What steps should be taken if SLOs are not being met to determine the problem?

## Implementation History

<!--
Major milestones in the lifecycle of a KEP should be tracked in this section.
Major milestones might include:
- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
- the `Proposal` section being merged, signaling agreement on a proposed design
- the date implementation started
- the first Kubernetes release where an initial version of the KEP was available
- the version of Kubernetes where the KEP graduated to general availability
- when the KEP was retired or superseded
-->

> ## Drawbacks

Reviewer: Probably this section should be filled out eventually.

Author: I've added a few obvious ones. More are welcome!

marked as `provisional` is a working document and subject to change. You can
denote sections that are under active debate as follows:

```
<<[UNRESOLVED optional short context or usernames ]>>
Stuff that is being argued.
<<[/UNRESOLVED]>>
```

When editing KEPS, aim for tightly-scoped, single-topic PRs to keep discussions
focused. If you disagree with what is already in a document, open a new PR
with suggested changes.

One KEP corresponds to one "feature" or "enhancement" for its whole lifecycle.
You do not need a new KEP to move from beta to GA, for example. If
new details emerge that belong in the KEP, edit the KEP. Once a feature has become
"implemented", major changes should get new KEPs.

The canonical place for the latest set of instructions (and the likely source
of this file) is [here](/keps/NNNN-kep-template/README.md).

**Note:** Any PRs to move a KEP to `implementable`, or significant changes once
it is marked `implementable`, must be approved by each of the KEP approvers.
If none of those approvers are still appropriate, then changes to that list
should be approved by the remaining approvers and/or the owning SIG (or
SIG Architecture for cross-cutting KEPs).
-->

> # KEP-3344: Refine Release Team Selection Process

Author: Done. I've updated everything, including the file location.

  This inhibits the ability of contributors to gain comfort and skill in a particular role. 
- Wider community members interact with a constantly changing team of people for every release, which makes it harder for them to build working relationships with the release team.
- With a few notable exceptions, once a contributor ends their time on the release team they don't come back.
  This "up or out" process makes it hard to do release team work as a long-standing community role.

There are a number of qualities to the current process that are desirable, and should be kept:

- The shadow survey is an excellent "on ramp" into the community, providing contributors with no experience a path to [org membership](https://github.com/kubernetes/community/blob/master/community-membership.md) and it is expected that contributors may only join the release team for a release or two before moving to other parts of the community.
- The release team having some degree of 'churn' is natural and expected.
  An entirely static, or effectively static, team is not desirable. 

Balancing these desires, this KEP introduces a long-lived "roster" of release team members.

> This roster is used to assemble release teams instead of a new shadow survey each time.

Reviewer 2: I think we should not replace the shadow survey for new folks who wanna join, so let's clarify:
```suggestion
This roster is used to assemble release teams with recurring contributors in addition to the shadow survey for new folks.
```

know that this has succeeded?
-->

- Continue to provide an "on ramp" for new community members.
- Provide a more streamlined process for release team shadows, including easier return into future release teams.
- Provide a "stable" position that a member may continue to be on release teams if they wish.
- Better handle contributors who wish to take breaks from the team, and then return. 

### Non-Goals

<!--
What is out of scope for this KEP? Listing non-goals helps to focus discussion
and make progress.
-->

- Substantively change the makeup and roles of the release team itself.
- Change how the release team lead, emeritus adviser, and role leads are selected.

## Proposal

<!--
This is where we get down to the specifics of what the proposal actually is.
This should have enough detail that reviewers can understand exactly what
you're proposing, but should not include things like API designs or
implementation. What is the desired outcome and how do we measure success?.
The "Design Details" section below is for the real
nitty-gritty.
-->

### Release Team Roster

We will add to SIG Release a "Release Team Roster".

> The list of roster members will be maintained in a YAML file in https://github.com/kubernetes/sig-release under a new `release-team/roster` directory.

Reviewer 2: I'd say we keep to k8s.io and add it as yaml like this: https://github.com/kubernetes/k8s.io/blob/c4b73a3e677f83091a2ab939b289d2f84aad9ff7/groups/sig-release/groups.yaml#L307-L363

> People on the roster can be in one of three roles: Shadow, Member, or Emeritus.

Reviewer 2: How about keeping the shadows as-is and only use "Member" or "Emeritus"? This way shadows can join the roster after a cycle, which provides a contributor ladder.

At the time of creation, all former role leads or release leads from any release will be added as emeritus members, or a member if they want.

> The 1.25 EA will lead discussions around which shadows from 1.25 should be placed at the member or shadow level on the release team roster.

Reviewer 2: nit: let's avoid building language barriers:
```suggestion
The 1.25 Emeritus Advisor (EA) will lead discussions around which shadows from 1.25 should be placed at the member or shadow level on the release team roster.
```

Roster Shadows may leave at any time by opening a PR to remove themselves.
Shadows that do not join a release team for two consecutive releases will generally be removed (although this is at the discretion of SIG Release).
Members may leave at any time by opening a PR to move themselves to the Emeritus section.

### Release Team Changes

The "Lead Shadow" role will be renamed to "Assistant Lead".
There is otherwise no change in the nature of this role.
This is to avoid confusion as to what a "shadow" is on the release team, as a "lead shadow" is significantly different to a role shadow.

### Shadow Survey

> The per-release shadow selection survey will be replaced with a rolling "expression of interest" form.

Reviewer 2: I think we should do both: Selecting new shadows and re-evaluating the existing roster. This makes it easier for new contributors to join because the folks from the roster will not use the shadow form any more.
